### PR TITLE
Update Insight search table column definitions

### PR DESCRIPTION
**Pull request by zlayne**
**Related Issue:**  open-cluster-management/backlog#11701

### Description of changes
- Updated the table column definitions for Insight resources to look like:
Screen Shot 2021-04-28 at 2 26 02 PM

/hold

### CODE REVIEW

**Comment by zlayne** [comment]
Hey @imjoyjean wanted to get your input on these changes. Our implementation of the PolicyReport resource has changed since we last spoke. We are now going to use 1 PolicyReport per cluster that will contain all the violations that cluster has (this will help on memory cost). To that end, we have to update the search table to reflect this change. I have included a screen cap of the changes as they stand now. 

The tooltip texts are as follows:
For Total Insight Policies - 
`This field is indexed as numInsightPolicies. If you would like to filter PolicyReports by number of violations please use: numInsightPolicies > x`

For Insight policies - 
`This field is indexed as insightPolicies. If you would like to filter PolicyReports containing specific policies please use: insightPolicies:<policyName>`

Please let me know if any of the text doesnt make sense / could be worded differently and if there should be anything added/removed from the table. Feel free to slack me as well.
Thanks!

**Comment by imjoyjean** [comment]
Nice! I think that makes sense to me in that you'll be able to see all the insight policies from a single cluster. Would we be able to tell n/total insights that are currently being violated? 

FYI @cadawson can you take a quick look and see if there's any thoughts you have anything you want to tweak on the tooltips that Zack provided us above?

**Comment by zlayne** [comment]
@imjoyjean With this new implementation of the PolicyReport it will only contain insights that are currently being violated. so the number in `Total insight policies` column will be the `total insights that are currently being violated` if that makes sense

**Comment by imjoyjean** [comment]
Could we combine those two columns and call it "Total insight policy violations" 
Name of table column - **Total insight policy violations** 
Value - **13**: policy1, policy 2, [10 more]

**Comment by cadawson** [comment]
Thanks for reaching out @imjoyjean and @zlayne . Some thoughts:

I agree that it looks like both columns might be redundant, unless I am missing something.  

When I looked at it, I wondered about the possibility of having it as one column:

Name of table column - Insight policy violations
Total - 13: policy1, policy 2, [10 more]

My thought there is that "Value" might not be as descriptive of what it could be, and moving "Total" from the top might give you some extra space there. That is up to you, though. 

For the tooltip, I would recommend something like: 

Total number of insight policies violated by the cluster. To view the total number of violations, search for `numInsightPolicies > 0`. To search for PolicyReports that contain specific policies, search for `insightPolicies`.

CD: Isn't the total number of violations already listed there in Total? I changed x to zero, as you are searching for any violations, I think?
I might have provided more questions than answers, but I might not follow the path. Maybe we can talk tomorrow, if you want. 

Thanks!

**Comment by imjoyjean** [comment]
Agreed with the changes - I also realized that we may not need the word "Total" because we'd always be showing all the violations (so i feel like "Total" is implied). 

---------

Name of table column - **Insight policy violations**
Value - **13**: policy1, policy 2, [10 more]

Tooltip: 
Total number of insight policies violated by the cluster. To view the total number of violations, search for numInsightPolicies > 0. To search for PolicyReports that contain specific policies, search for insightPolicies.

**Comment by zlayne** [comment]
@cadawson @imjoyjean Thanks for all the feedback. I'm on board with the suggestions, one small comment though. 
A PolicyReport will never have 0 violations. If the cluster does not have any violations we wont create a PolicyReport for the cluster, so it will always be 1 or more. 

Bringing that up as the tooltip text might need to change slightly. Maybe something like:

`Total number of insight policies violated by the cluster. To view clusters based on number of violations, search for numInsightPolicies. To search for PolicyReports that contain specific policies, search for insightPolicies`

**Comment by zlayne** [comment]
@imjoyjean Now that I am implementing these changes I am wondering if having the total count before the policy list is necessary? There is a button at the end of the list which shows the number of remaining policies:

Screen Shot 2021-04-28 at 6 35 46 PM

**Comment by imjoyjean** [comment]
Good point, removing the total count makes sense to me now that I see it. Also, could we switch out the pills and make it plain text? An example of when we do that is in GRC -> image

File: frontend/src/routes/SearchPage/searchDefinitions.tsx [hunk 1]
```diff
@@ -1031,21 +1032,17 @@ const searchDefinitions: any = {
...
-            {
-                header: 'Total Risk',
-                sort: 'risk',
-                cell: 'risk',
+                header: 'Insight policy violations',
+                cell: (item: any) => {
+                    return FormatInsightPolicies(item)
+                },
+                tooltip: `Total number of insight policies violated by the cluster. To view clusters based on number of violations,
+                     search for numInsightPolicies. To search for PolicyReports that contain specific policies, search for insightPolicies.`,
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/searchDefinitions.tsx:1040]: Move strings to translation file.

File: frontend/src/routes/SearchPage/searchDefinitions.tsx [hunk 2]
```diff
@@ -1450,10 +1447,43 @@ export function FormatLabels(item: any) {
...
+    }
+    return '-'
+}
+
+export function FormatInsightPolicies(item: any) {
+    if (item.insightPolicies) {
+        const policyArray = item.insightPolicies.split('; ')
+        const policiesToHide = policyArray.slice(2)
+        return (
+            <LabelGroup collapsedText={`${policiesToHide.length} more`} expandedText={'Show less'} numLabels={2}>
```
Comment (jlpadilla) [inline frontend/src/routes/SearchPage/searchDefinitions.tsx:1459]: Move strings to translation file.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

Minor comment for display text strings.

**Comment by cadawson** [comment]
Good call with removing the total number. As long as the `+11 more` is there, they should be able to figure out how many total there are easily.

**Pull request merged**